yb-voyager/src/dbzm/valueConverter.go

```diff
@@ -0,0 +1,142 @@
+
+type DebeziumValueConverter struct {
+	schemaRegistry      *SchemaRegistry
+	valueConverterSuite map[string]tgtdb.ConverterFn
+	converterFnCache    map[string][]tgtdb.ConverterFn
+	snapshotMode        bool
+}
+
+func NewDebeziumValueConverter(exportDir string, tdb tgtdb.TargetDB, snapshotMode bool) *DebeziumValueConverter {
+	schemaRegistry := NewSchemaRegistry(exportDir)
+	tdbValueConverterSuite := tdb.GetDebeziumValueConverterSuite(snapshotMode)
+
+	return &DebeziumValueConverter{
+		schemaRegistry:      schemaRegistry,
+		valueConverterSuite: tdbValueConverterSuite,
+		converterFnCache:    map[string][]tgtdb.ConverterFn{},
+		snapshotMode:        snapshotMode,
+	}
+}
+
+func (conv *DebeziumValueConverter) ConvertRow(tableName string, columnNames []string, row string) (string, error) {
+	converterFns, err := conv.getConverterFnCache(tableName, columnNames)
+	if err != nil {
+		return row, err
+	}
+	columnValues := strings.Split(row, "\t")
+	var transformedValues []string
+	for i, columnValue := range columnValues {
+		if columnValue != "\\N" {
+			transformedValue, err := converterFns[i](columnValue)
+			if err != nil {
+				return row, err
+			}
+			transformedValues = append(transformedValues, transformedValue)
+		} else {
+			transformedValues = append(transformedValues, columnValue)
+		}
+	}
+	transformedRow := strings.Join(transformedValues, "\t")
+	return transformedRow, nil
+}
+
+func (conv *DebeziumValueConverter) getConverterFnCache(tableName string, columnNames []string) ([]tgtdb.ConverterFn, error) {
+	result := conv.converterFnCache[tableName]
+	if result == nil {
+		colSchemas, err := conv.schemaRegistry.GetColumnSchemas(tableName, columnNames)
+		if err != nil {
+			return nil, err
+		}
+		result = make([]tgtdb.ConverterFn, len(columnNames))
+		for i := range columnNames {
+			schema := colSchemas[i]
+			result[i] = schema.ColDbzSchema.GetConverterFn(conv.valueConverterSuite)
+		}
+		conv.converterFnCache[tableName] = result
+	}
+	return result, nil
+}
+
+func (conv *DebeziumValueConverter) ConvertEvent(ev *tgtdb.Event, table string) error {
+	err := conv.convertMap(table, ev.Key)
+	if err != nil {
+		return fmt.Errorf("convert event fields: %v", err)
+	}
+	err = conv.convertMap(table, ev.Fields)
+	if err != nil {
+		return fmt.Errorf("convert event keys: %v", err)
+	}
+	return nil
+}
+
+func (conv *DebeziumValueConverter) convertMap(tableName string, m map[string]interface{}) error {
+	for column, value := range m {
+		if value == nil {
+			m[column] = "NULL"
+			continue
+		}
+		columnValue := fmt.Sprintf("%v", value)
+		colSchema, err := conv.schemaRegistry.GetColumnSchema(tableName, column)
+		if err != nil {
+			return fmt.Errorf("fetch column schema: %v",err)
+		}
+		converterFn := colSchema.ColDbzSchema.GetConverterFn(conv.valueConverterSuite)
+		columnValue, err = converterFn(columnValue)
+		if err != nil {
+			return fmt.Errorf("converting value: %v",err)
```

Comment:
Provide enough information in the error context so that one can start debugging e.g.
```
return fmt.Errorf("error while converting %s.%s of type %s in event[%d]: %w", tableName, columnName, typeName, event.SequenceID, err)
```

Reply:
Is there any SequenceId for each event?